RestClient.cpp

```diff
@@ -326,18 +308,43 @@ int RestClient::readResponse(String* response) {
                     }
 
                     if (c == '\n') {
-                        // you're starting a new line
+                        // your starting a new line
                         currentLineIsBlank = true;
                     }
                     else if (c != '\r') {
                         // you've gotten a character on the current line
                         currentLineIsBlank = false;
                     }
+
+                	rawLastResponse = rawLastResponse + c;
                 }
             }
         }
+        while (client.available()) {
+            char c = client.read();
+            if (httpBody)
+                if (response)
+                    response->concat(c);
+        }
     }
 
     HTTP_DEBUG_PRINT("HTTP: return readResponse3\n");
     return code;
 }
+
+String RestClient::getRawLastResponse() {
+	return rawLastResponse;
+}
+
+void RestClient::getResponseHeader(const char* key, String* value) {
+	HTTP_DEBUG_PRINT(rawLastResponse);
+	String fullLengthPrefix = (String) key + ": ";
+	HTTP_DEBUG_PRINT(fullLengthPrefix);
+	int start = rawLastResponse.indexOf(fullLengthPrefix);
+	int end = rawLastResponse.indexOf("\n", start);
```

Comment:
According to the spec https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html the end of the end of the headers should be `\r\n\r\n`